theme/static/js/hs_file_browser.js

```diff
@@ -1080,19 +1080,14 @@ $(document).ready(function () {
 
     function onPaste() {
         var folderName = $("#fb-files-container li.ui-selected").children(".fb-file-name").text();
-        var targetPath = $("#hs-file-browser").attr("data-current-path");
-
-        if (folderName && folderName.lastIndexOf(".") == -1) {  // Makes sure the destination is a folder
```

Comment:
For the implementation of Paste in the drop method see https://github.com/hydroshare/hydroshare/blob/5a20725e54f1e7e9f8c61b33d6afe0f27d5b5d64/theme/static/js/hs_file_browser.js#L239

Note that here the erroneous condition with dots in the folder name doesn't exist here and if this is how the error has been reproduced then the cause might be somewhere else.